packages/types/src/multiDomain.ts

```diff
@@ -7,19 +7,46 @@ import type { ClerkOptions } from './clerk';
  * 3) isSatellite and proxy is set
  * 4) isSatellite and domain is set
  */
-export type DomainOrProxyUrl =
+export type MultiDomainAndOrProxy =
+  | {
+      isSatellite?: never;
+      proxyUrl?: never | string;
+      domain?: never;
+    }
+  | {
+      isSatellite: Exclude<ClerkOptions['isSatellite'], undefined>;
+      proxyUrl?: never;
+      domain: Exclude<ClerkInterface['domain'], undefined>;
+    }
+  | {
+      isSatellite: Exclude<ClerkOptions['isSatellite'], undefined>;
```

Comment:
❓ Wouldn't it be more readable to move the `isSatellite` from `ClerkOptions` to here  and avoid `Exclude`  ? Also semantically if `isSatellite` only used as part of multi-domain feature i would expect to find it as part of the `MultiDomainAndOrProxy` type.
Also i would rename the type from `MultiDomainAndOrProxy` to `MultiDomainAndProxy`. WDYT ?

Reply:
Could you explain what are the validations of this type? What are the combination 😓 ?

Reply:
Yes, I've listed all valid and invalid combos in a [previous PR](https://github.com/clerkinc/javascript/pull/901)

Reply:
Regarding your 1st comment

isSatellite lives inside ClerkOptions because it is used by `new Clerk().load()`

The patterns that existed before ( and I followed ) is: Pass to the Clerk Class construct only the information necessary to create the instance (e.g createFapiClient)

And pass everything else through `load` as `ClerkOptions`

We can for sure refactor this. I'm just laying out my reasoning behind this